File: server/mc/common2/src/main/java/calebxzhou/rdi/mc/common2/player/RGlobalPlayerList.java
package calebxzhou.rdi.mc.common2.player;

import java.util.List;


public record RGlobalPlayerList(long generatedAt, List<HostEntry> hosts) {
    public RGlobalPlayerList {
        hosts = hosts == null ? List.of() : List.copyOf(hosts);
    }

    public record HostEntry(
            String hostId,
            String hostName,
            String modpackName,
            String packVer,
            List<PlayerEntry> players
    ) {
        public HostEntry {
            players = players == null ? List.of() : List.copyOf(players);
        }
    }

    public record PlayerEntry(String playerId, String playerName) {
    }
}

File: server/master/src/main/kotlin/calebxzhou/rdi/master/model/RGlobalPlayerList.kt
package calebxzhou.rdi.master.model

import kotlinx.serialization.Serializable

@Serializable
data class RGlobalPlayerList(
    val generatedAt: Long,
    val hosts: List<HostEntry>,
) {
    @Serializable
    data class HostEntry(
        val hostId: String,
        val hostName: String,
        val modpackName: String,
        val packVer: String,
        val players: List<PlayerEntry>,
    )

    @Serializable
    data class PlayerEntry(
        val playerId: String,
        val playerName: String,
    )
}
